src/core/order_manager.py

```diff
@@ -1,19 +1,22 @@
 # order manager mechanism
 
 from queue import Queue
+from src.constants.order_status import OrderStatus
 
 
 class OrderManager:
     def __init__(self):
         self._order_status_to_order_queue_map = {
-            "order_placed": Queue(maxsize=0),
-            "in_process": Queue(maxsize=0),
-            "cancelled": Queue(maxsize=0),
-            "completed": Queue(maxsize=0),
+            OrderStatus.NEW_ORDER: Queue(maxsize=1000),
+            OrderStatus.IN_PROCESS: Queue(maxsize=1000),
+            OrderStatus.CANCELLED: Queue(maxsize=1000),
+            OrderStatus.COMPLETED: Queue(maxsize=1000),
         }
 
     def add_to_queue(self, order):
-        self._order_status_to_order_queue_map[order.status.value].put(order.id)
+        if self.get_queue_size(order.status) == 1000:
+            return
```

Comment:
Could you raise an error instead? You wouldn't like the program to continue. Let the dev decide whst to do with the error. They can just ignore or handle properly with try/catch.

Reply:
do. i added an raise overflow error